src/Interpreters/AsynchronousMetrics.cpp

```diff
@@ -674,21 +694,32 @@ void AsynchronousMetrics::update(std::chrono::system_clock::time_point update_ti
         {
             Int64 amount = total_memory_tracker.get();
             Int64 peak = total_memory_tracker.getPeak();
-            Int64 new_amount = data.resident;
+            Int64 rss = data.resident;
 
-            Int64 difference = new_amount - amount;
+#if USE_JEMALLOC
+            /// This is a memory which is kept by allocator.
+            /// Remove it from RSS to decrease memory drift.
+            rss -= je_malloc_mapped - je_malloc_allocated;
+#endif
+            /// In theory, the difference between RSS and tracked memory should be caused by
+            /// external libraries which allocation we can't track.
+            Int64 rss_drift = rss - amount;
+            Int64 difference = rss_drift - last_logged_rss_drift;
 
             /// Log only if difference is high. This is for convenience. The threshold is arbitrary.
             if (difference >= 1048576 || difference <= -1048576)
+            {
                 LOG_TRACE(log,
-                    "MemoryTracking: was {}, peak {}, will set to {} (RSS), difference: {}",
+                    "MemoryTracking: allocated {}, peak {}, RSS {}, difference: {}",
```

Comment:
I was wondering, maybe it worth adjusting the message? Since this is not RSS anymore, but something like `RSS without allocator fragmentation` (I'm not insisting on this particular words)